web/source/keyboards/kmwkeyboards.ts

```diff
@@ -238,23 +238,29 @@ namespace com.keyman.keyboards {
      *   This was defined as an array, so is kept that way, but
      *   Javascript treats it as an object anyway
      *
-     * @param       {Object}    Lkbd       Keyboard object
+     * @param       {Object}    Lstub      Keyboard stub object
+     * @param       {Object}    Lkbd       Keyboard script object
      * @return      {Object}               Copy of keyboard identification strings
      *
      */
-    private _GetKeyboardDetail = function(Lkbd) { // I2078 - Full keyboard detail
+    private _GetKeyboardDetail = function(Lstub: KeyboardStub, Lkbd: any /* KeyboardScriptObject */) { // I2078 - Full keyboard detail
       var Lr={};
-      Lr['Name'] = Lkbd['KN'];
-      Lr['InternalName'] =  Lkbd['KI'];
-      Lr['LanguageName'] = Lkbd['KL'];  // I1300 - Add support for language names
-      Lr['LanguageCode'] = Lkbd['KLC']; // I1702 - Add support for language codes, region names, region codes, country names and country codes
-      Lr['RegionName'] = Lkbd['KR'];
-      Lr['RegionCode'] = Lkbd['KRC'];
-      Lr['CountryName'] = Lkbd['KC'];
-      Lr['CountryCode'] = Lkbd['KCC'];
-      Lr['KeyboardID'] = Lkbd['KD'];
-      Lr['Font'] = Lkbd['KFont'];
-      Lr['OskFont'] = Lkbd['KOskFont'];
+      Lr['Name'] = Lstub['KN'];
+      Lr['InternalName'] =  Lstub['KI'];
+      Lr['LanguageName'] = Lstub['KL'];  // I1300 - Add support for language names
+      Lr['LanguageCode'] = Lstub['KLC']; // I1702 - Add support for language codes, region names, region codes, country names and country codes
+      Lr['RegionName'] = Lstub['KR'];
+      Lr['RegionCode'] = Lstub['KRC'];
+      Lr['CountryName'] = Lstub['KC'];
+      Lr['CountryCode'] = Lstub['KCC'];
+      Lr['KeyboardID'] = Lstub['KD'];
+      Lr['Font'] = Lstub['KFont'];
+      Lr['OskFont'] = Lstub['KOskFont'];
+      Lr['HasLoaded'] = !!Lkbd;
+
+      if(Lkbd) {
+        Lr['IsRTL'] = !!Lkbd['KRTL'];
+      }
```

Comment:
```suggestion
      Lr['IsRTL'] = Lkbd ? !!Lkbd['KRTL'] : null;
```

Two things:

1. This way `IsRTL` is still an iterable property even if it has an `null` value. Just prettier.
2. I prefer `null` to `undefined` here because although `undefined` *sounds* appropriate -- we haven't yet defined the value -- it is a weird quirky thing and `null` is a better tri-state "undefined" value... ([`undefined` shouldn't exist!](https://stackoverflow.com/a/2236186/1836776))